components/fxa-client/src/ffi.rs

```diff
@@ -276,6 +277,38 @@ unsafe fn get_buffer<'a>(data: *const u8, len: i32) -> &'a [u8] {
     }
 }
 
+impl From<msg_types::AuthorizationParams> for AuthorizationParameters {
+    fn from(proto_prams: msg_types::AuthorizationParams) -> Self {
+        Self {
+            client_id: proto_prams.client_id,
+            scope: proto_prams.scope,
+            state: proto_prams.state,
+            access_type: proto_prams.access_type,
+            key_params: proto_prams.key_params.map(|key_params| key_params.into()),
+        }
+    }
+}
+
+impl From<msg_types::AuthorizationKeyParams> for AuthorizationKeyParams {
+    fn from(proto_key_params: msg_types::AuthorizationKeyParams) -> Self {
+        Self {
+            code_challenge: proto_key_params.code_challenge,
+            code_challenge_method: proto_key_params.code_challenge_method,
+            keys_jwk: proto_key_params.keys_jwk,
+        }
+    }
+}
+
+impl AuthorizationParameters {
+    /// # Safety
+    /// Deref pointer thus unsafe
```

Comment:
Heh, I know you copied this from elsewhere in our code, but I laugh when I see these comments. IIUC clippy lints on these to remind us to document the things that callers of the function need to do to ensure safety, but our comments are just like "Yep, it's unsafe! Got some pointers in here!".

(I'm not asking you to do anything about it, just noting my amusement 😄)